awx/main/migrations/0076_v360_add_new_instance_group_relations.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-05-06 15:20
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from awx.main.fields import OrderedManyToManyField
```

Comment:
Question:

Let me play out the future in my head.

We merge this
Someone starts upgrading to Django 2.0
On startup errors are thrown, the add and remove methods are deleted from `OrderedManyToManyDescriptor`
then `OrderedManyToManyDescriptor` may or may not remain necessary, depending on if Django will let us order the through model

Do we still keep `OrderedManyToManyField`?

We're importing the field here, so if we deleted the field class, then we would have to change the migration.

But it looks more like, for the Django 2.0 upgrade, we just delete some of the most painful parts of the hack, but the ordered M2M field will remain a thing.

Reply:
Yep, my thought is we keep this ordered field - even in Django 2, we still need it to apply the default sort order.

At upgrade time, we just lose the custom `.add()` and `.remove()` methods.

Reply:
I have to admit, I don't love it.

Reply:
love is the answer